proto/data.go

```diff
@@ -316,6 +317,69 @@ func (v *Value) SetProto(msg gogoproto.Message) error {
 	return nil
 }
 
+// SetTime encodes the specified time value into the bytes field of the
+// receiver and sets the tag.
+func (v *Value) SetTime(t time.Time) error {
+	var err error
+	if v.Bytes, err = t.MarshalBinary(); err != nil {
+		return err
+	}
+	v.Tag = ValueType_TIME
+	return nil
+}
+
+// GetBytesChecked retrieves the bytes value from receiver returning an error
+// if the tag is not BYTES.
+//
+// TODO(pmattis): This method should be named GetBytes() but that method is
+// auto-generated by protoc. Perhaps rename the Bytes field to avoid the
+// collision here.
+func (v *Value) GetBytesChecked() ([]byte, error) {
+	if v == nil || v.Tag != ValueType_BYTES {
+		return nil, fmt.Errorf("value type is not BYTES: %s", v.GetTag())
+	}
+	return v.Bytes, nil
+}
+
+// GetFloat decodes a float64 value from the bytes field of the receiver. If
+// the bytes field is not 8 bytes in length or the tag is not FLOAT an error
+// will be returned.
+func (v *Value) GetFloat() (float64, error) {
+	if v == nil || v.Tag != ValueType_FLOAT {
+		return 0, fmt.Errorf("value type is not FLOAT: %s", v.GetTag())
+	}
+	if len(v.Bytes) != 8 {
+		return 0, fmt.Errorf("float64 value should be exactly 8 bytes: %d", len(v.Bytes))
+	}
+	_, u := encoding.DecodeUint64(v.Bytes)
+	return math.Float64frombits(u), nil
+}
+
+// GetInt decodes an int64 value from the bytes field of the receiver. If the
+// bytes field is not 8 bytes in length or the tag is not INT an error will be
+// returned.
+func (v *Value) GetInt() (int64, error) {
+	if v == nil || v.Tag != ValueType_INT {
+		return 0, fmt.Errorf("value type is not INT: %s", v.GetTag())
```

Comment:
i think you can simplify the check to `if tag := v.GetTag(); tag != ValueType_INT {` because `GetTag()` will return `ValueType_UNKNOWN` if v is nil